src/wallet/wallet.cpp

```diff
@@ -3891,11 +3902,16 @@ bool CWallet::ApplyMigrationData(MigrationData& data, bilingual_str& error)
         if (ExtractDestination(script, dest)) not_migrated_dests.emplace(dest);
     }
 
+    Assume(m_cached_spks.empty());
+
     for (auto& desc_spkm : data.desc_spkms) {
         if (m_spk_managers.count(desc_spkm->GetID()) > 0) {
             error = _("Error: Duplicate descriptors created during migration. Your wallet may be corrupted.");
             return false;
         }
+        for (const auto& script : desc_spkm->GetScriptPubKeys()) {
+            m_cached_spks[script].insert(desc_spkm.get());
+        }
```

Comment:
nit: sounds like these two changes should be in the previous commit (cde66ceae).

Reply:
Done